### ISSUE TITLE
URLs in the sitemap don't respect default_url_options.

### ISSUE THREAD

**Issue by tosbourn** [kjvarga/sitemap_generator#355]
With a Rails project setup with the following line in `config/application.rb`

```
config.action_controller.default_url_options = { trailing_slash: true }
```

You would expect `foos_path` to generate `/foos/`

When referencing `foos_path` from `config/sitemap.rb` the path that is generated is the default Rails one, `/foos` (without the trailing slash).

The workaround is to call `foos_path(trailing_slash: true)`, but it would be nice if the sitemap respected the options laid out in for `default_url_options`, even though these are bundled in `action_controller`.

**Comment by dtbaker** [comment]
@tosbourn did you figure out a way to do this globally? I just ran into the same issue. Thanks for the `trailing_slash: true` hunt, I'll do that until I can find a way to do it globally

**Comment by tosbourn** [comment]
@dtbaker nope, I've just been adding `trailing_slash: true`, now the code has a president for it, there isn't a big cognitive burden to doing it on new additions to the file.

**Comment by jasonkarns** [comment]
The railtie is getting some enhancements (#478 ). It's possible now that the railtie could save off the detected default_url_options and allow Sitemap to use them during generation?

**Comment by kjvarga** [comment]
Fixed in **v7.1.1**, released on RubyGems: https://rubygems.org/gems/sitemap_generator/versions/7.1.1

`SitemapGenerator::Interpreter` now defines `default_url_options`, so Rails URL helpers used inside `config/sitemap.rb` pick up `ActionController::Base.default_url_options` (e.g. `:protocol`, `:port`, `:trailing_slash`) instead of ignoring them.

Fixed via #507. See the [7.1.1 changelog entry](https://github.com/kjvarga/sitemap_generator/blob/v7.1.1/CHANGES.md#711).


*Generated by Claude*
